pkg/cmd/label/list/list.go

```diff
@@ -0,0 +1,124 @@
+package list
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmd/label/shared"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/cli/cli/v2/utils"
+	"github.com/spf13/cobra"
+)
+
+type browser interface {
+	Browse(string) error
+}
+
+type ListOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+	Browser    browser
+
+	WebMode bool
+}
+
+func NewCmdList(f *cmdutil.Factory, runF func(*ListOptions) error) *cobra.Command {
+	opts := ListOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		BaseRepo:   f.BaseRepo,
+		Browser:    f.Browser,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   "Show all labels",
+		Long:    "Display all labels of a GitHub repository.",
+		Args:    cobra.NoArgs,
+		Aliases: []string{"ls"},
+		RunE: func(c *cobra.Command, args []string) error {
+			if runF != nil {
+				return runF(&opts)
+			}
+			return listRun(&opts)
+		},
+	}
+
+	cmd.Flags().BoolVarP(&opts.WebMode, "web", "w", false, "List labels in the web browser")
+
+	return cmd
+}
+
+func listRun(opts *ListOptions) error {
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+
+	baseRepo, err := opts.BaseRepo()
+	if err != nil {
+		return err
+	}
+
+	if opts.WebMode {
+		labelListURL := ghrepo.GenerateRepoURL(baseRepo, "labels")
+
+		if opts.IO.IsStdoutTTY() {
+			fmt.Fprintf(opts.IO.ErrOut, "Opening %s in your browser.\n", utils.DisplayURL(labelListURL))
+		}
+
+		return opts.Browser.Browse(labelListURL)
+	}
+
+	client := api.NewClientFromHTTP(httpClient)
+
+	opts.IO.StartProgressIndicator()
+
+	labels, err := getLabels(client, baseRepo)
+	if err != nil {
+		return err
+	}
+
+	opts.IO.StopProgressIndicator()
+
+	printLabels(labels, opts.IO)
+
+	return nil
+}
+
+func getLabels(client *api.Client, repo ghrepo.Interface) ([]shared.Label, error) {
```

Comment:
This does not handle the case where there is more than one page of labels. For example using this `gh label list -R cli/cli` does not list all the labels that are available because we have so many. I believe the other `list` commands should have examples of how to handle this type of pagination. Additionally, due to the possibility of a repo having thousands of labels we should introduce a `--limit` flag that will act as a stopper once a certain number of labels have been retrieved.